pkgs/test/test/runner/json_reporter_test.dart

```diff
@@ -598,207 +598,5 @@ import 'package:test/test.dart';
   await test.shouldExit();
 
   var stdoutLines = await test.stdoutStream().toList();
-  // Ensure the output is of the same length, including start, done and all
-  // suites messages.
-  expect(stdoutLines.length, equals(expected.fold(3, (a, m) => a + m.length)),
-      reason: 'Expected $stdoutLines to match $expected.');
-
-  // TODO(nweiz): validate each event against the JSON schema when
-  // patefacio/json_schema#4 is merged.
-  dynamic decodeLine(String l) =>
-      jsonDecode(l)..remove('time')..remove('stackTrace');
-
-  // Should contain all suites message.
-  expect(stdoutLines.map(decodeLine), containsAll([_allSuites()]));
-
-  // A single start event is emitted first.
-  final _start = {
-    'type': 'start',
-    'protocolVersion': '0.1.1',
-    'runnerVersion': testVersion,
-    'pid': test.pid
-  };
-  expect(decodeLine(stdoutLines.first), equals(_start));
-
-  // A single done event is emmited last.
-  expect(decodeLine(stdoutLines.last), equals(done));
-
-  for (var value in expected) {
-    expect(stdoutLines.map(decodeLine), containsAllInOrder(value));
-  }
-}
-
-/// Returns the event emitted by the JSON reporter providing information about
-/// all suites.
-///
-/// The [count] defaults to 1.
-Map _allSuites({int count}) {
```

Comment:
I moved all of these out to a separate file so that they could be shared between this test and the file reporter test.

Reply:
If that's too much moved code for not a good enough reason, I could also move the file reporter tests into this test file.

Reply:
The move looks good to me.